### PR TITLE
pixelmod interface design

### PR DESCRIPTION
**Pull request by DaviesX**
1. Created pixelmod gradle package.
2. Added interface to render decal and builder class that creates decal
renderer.

### CODE REVIEW

**Comment by zhaonian** [comment]
CircleCI complains about lint error. make sure to fix all the problems raised by running these:
`./gradlew lint` - for general android lint
`./gradlew ktlint` - for general kotlint lint
`./gradlew detekt` - for kotlin static analysis

File: pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package io.pixelmod
+
+import android.graphics.Bitmap
+
+interface IDecalRenderer {
```
Comment (zhaonian) [inline pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt:5]: I don't like the `I` prefix. feels like C++

File: pixelmod/src/main/java/io/pixelmod/DecalRendererBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package io.pixelmod
+
+enum class DecalRendererAlgorithmType {
+    SIFT_DECAL_RENDERER
+}
+
+class DecalRendererBuilder {
```
Comment (zhaonian) [inline pixelmod/src/main/java/io/pixelmod/DecalRendererBuilder.kt:9]: Kotlin! nice! 🎉

File: pixelmod/build.gradle [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    // Necessary for Robolectric unit tests
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
+}
+
+dependencies {
```
Comment (zhaonian) [inline pixelmod/build.gradle:21]: You probably only need these dependencies:
`Kotlin`,
`App compat`,
`OpenCV Android`,
`Unit testing`

**Review by zhaonian — APPROVED** [review]

**Review by zhaonian — CHANGES_REQUESTED** [review]
wait. let me modify something :)

File: pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt (hunk 1 above)
Comment (DaviesX) [reply pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt:5]: This is pretty common in any OOP languages.

[example](https://github.com/AndroidSDKSources/android-sdk-sources-for-api-level-28/blob/809dd8b7822332cdbc68939d777eb70a6951908a/android/content/IContentProvider.java)

**Review by zhaonian — APPROVED** [review]

File: pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt (hunk 1 above)
Comment (zhaonian) [reply pixelmod/src/main/java/io/pixelmod/IDecalRenderer.kt:5]: hehe

**Pull request merged**